setuptools/_distutils/command/install_data.py

```diff
@@ -56,6 +57,16 @@ def run(self):
                     )
                 (out, _) = self.copy_file(f, self.install_dir)
                 self.outfiles.append(out)
+            elif isinstance(f, Path):
+                # it's a simple file, so copy it
+                f = convert_path(str(f))
+                if self.warn_dir:
+                    self.warn(
+                        "setup script did not provide a directory for "
+                        f"'{f}' -- installing right in '{self.install_dir}'"
+                    )
+                (out, _) = self.copy_file(f, self.install_dir)
+                self.outfiles.append(out)
```

Comment:
Looking at this approach, I think I'm going to decline. I want to avoid creating more branches and duplicating more code.